source/server/connection_handler_impl.cc

```diff
@@ -72,23 +85,30 @@ void ConnectionHandlerImpl::ActiveListener::removeConnection(ActiveConnection& c
   parent_.num_connections_--;
 }
 
-ConnectionHandlerImpl::ActiveListener::ActiveListener(ConnectionHandlerImpl& parent,
-                                                      Network::ListenerConfig& config)
-    : ActiveListener(
+ConnectionHandlerImpl::ActiveListenerBase::ActiveListenerBase(ConnectionHandlerImpl& parent,
+                                                              Network::ListenerPtr&& listener,
+                                                              Network::ListenerConfig& config)
+    : parent_(parent), listener_(std::move(listener)),
+      stats_(generateStats(config.listenerScope())),
+      listener_filters_timeout_(config.listenerFiltersTimeout()),
+      listener_tag_(config.listenerTag()), config_(config) {}
+
+ConnectionHandlerImpl::ActiveListenerBase::~ActiveListenerBase() {}
```

Comment:
del empty destructor

Reply:
I added it as its virtual - I can move it into the .h